testing/checkin/runner/mod.rs

```diff
@@ -0,0 +1,149 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+use std::path::Path;
+use std::rc::Rc;
+
+use anyhow::Error;
+use deno_core::url::Url;
+use deno_core::JsRuntime;
+use deno_core::PollEventLoopOptions;
+use deno_core::RuntimeOptions;
+use pretty_assertions::assert_eq;
+use testing::Output;
+use tokio::fs::File;
+use tokio::io::AsyncReadExt;
+
+use self::testing::TestFunctions;
+
+mod ops;
+mod testing;
+mod ts_module_loader;
+
+deno_core::extension!(
+  checkin_runtime,
+  ops = [
+    ops::op_log_debug,
+    ops::op_log_info,
+    ops::op_test_register,
+  ],
+  esm_entry_point = "ext:checkin_runtime/__init.js",
+  esm = [
+    dir "checkin/runtime",
+    "__init.js",
+    "console.ts" with_specifier "checkin:console",
+    "testing.ts" with_specifier "checkin:testing",
+  ],
+  state = |state| {
+    state.put(TestFunctions::default());
+    state.put(Output::default());
+  }
+);
+
+fn create_runtime() -> JsRuntime {
+  let mut extensions = vec![checkin_runtime::init_ops_and_esm()];
+
+  for extension in &mut extensions {
+    use ts_module_loader::maybe_transpile_source;
+    for source in extension.esm_files.to_mut() {
+      maybe_transpile_source(source).unwrap();
+    }
+    for source in extension.js_files.to_mut() {
+      maybe_transpile_source(source).unwrap();
+    }
+  }
+
+  JsRuntime::new(RuntimeOptions {
+    extensions,
+    module_loader: Some(Rc::new(
+      ts_module_loader::TypescriptModuleLoader::default(),
+    )),
+    get_error_class_fn: Some(&|error| {
+      deno_core::error::get_custom_error_class(error).unwrap()
+    }),
+    ..Default::default()
+  })
+}
+
+pub fn run_system_test(test: &str) {
+  let runtime = create_runtime();
+  let tokio = tokio::runtime::Builder::new_current_thread()
+    .enable_all()
+    .build()
+    .expect("Failed to build a runtime");
+  tokio
+    .block_on(run_system_test_task(runtime, test.to_owned()))
+    .expect("Failed to complete test");
+}
+
+async fn run_system_test_task(
```

Comment:
Please add a comment, I don't know what "system test" is :(

Reply:
Added a couple in here. A system tests runs a full file and compares the output to a given .out file. I will rename to integration_tests to match Deno.